File: src/_1ProgrammingBasicsWithJavaMay2023/_3ConditionalStatementsAdvanced/_1Lab/_12TradeCommissions.java
package _1ProgrammingBasicsWithJavaMay2023._3ConditionalStatementsAdvanced._1Lab;

import java.util.Scanner;

public class _12TradeCommissions {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String town = scanner.nextLine();
        double income = Double.parseDouble(scanner.nextLine());
        boolean isValidData = true;
        double commission = 0;
        if (town.equals("Sofia")) {
            if (income >=0 && income <= 500) {
                commission = income * 0.05;
            } else if (income > 500 && income <= 1000) {
                commission = income * 0.07;
            } else if (income > 1000 && income <= 10000) {
                commission = income * 0.08;
            } else if (income > 10000) {
                commission = income * 0.12;
            } else {
                isValidData = false;
            }
        } else if (town.equals("Varna")) {
            if (income >=0 && income <= 500) {
                commission = income * 0.045;
            } else if (income > 500 && income <= 1000) {
                commission = income * 0.075;
            } else if (income > 1000 && income <= 10000) {
                commission = income * 0.10;
            } else if (income > 10000) {
                commission = income * 0.13;
            } else {
                isValidData = false;
            }
        } else if (town.equals("Plovdiv")) {
            if (income >=0 && income <= 500) {
                commission = income * 0.055;
            } else if (income > 500 && income <= 1000) {
                commission = income * 0.08;
            } else if (income > 1000 && income <= 10000) {
                commission = income * 0.12;
            } else if (income > 10000) {
                commission = income * 0.145;
            } else {
                isValidData = false;
            }
        } else {
            isValidData = false;
        }
        if (isValidData) {
            System.out.printf("%.2f",commission);
        } else {
            System.out.println("error");
        }
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_3ConditionalStatementsAdvanced/_1Lab/_12TradeCommissions.js
function tradeCommissions(city, sales) {
    let commissionPercentage = 0;

    if (city === "Sofia") {
        if (sales > 10000) {
            commissionPercentage = 0.12;
        } else if (sales > 1000) {
            commissionPercentage = 0.08;
        } else if (sales > 500) {
            commissionPercentage = 0.07;
        } else if (sales >= 0) {
            commissionPercentage = 0.05;
        } else {
            console.log(`error`);
            return;
        }
    } else if (city === "Varna") {
        if (sales > 10000) {
            commissionPercentage = 0.13;
        } else if (sales > 1000) {
            commissionPercentage = 0.10;
        } else if (sales > 500) {
            commissionPercentage = 0.075;
        } else if (sales >= 0) {
            commissionPercentage = 0.045;
        } else {
            console.log(`error`);
            return;
        }
    } else if (city === "Plovdiv") {
        if (sales > 10000) {
            commissionPercentage = 0.145;
        } else if (sales > 1000) {
            commissionPercentage = 0.12;
        } else if (sales > 500) {
            commissionPercentage = 0.08;
        } else if (sales >= 0) {
            commissionPercentage = 0.055;
        } else {
            console.log(`error`);
            return;
        }
    } else {
        console.log(`error`);
        return;
    }

    const commission = commissionPercentage * sales;

    console.log(`${commission.toFixed(2)}`);
}